Tests/AsyncQueueTests/FIFOQueueTests.swift

```diff
@@ -65,7 +91,54 @@ final class FIFOQueueTests: XCTestCase {
         await systemUnderTest.await { /* Drain the queue */ }
     }
 
+    func test_asyncOn_executesAsyncBlocksAtomically() async {
+        let semaphore = Semaphore()
+        for _ in 1...1_000 {
+            systemUnderTest.async(on: semaphore) { semaphore in
+                let isWaiting = semaphore.isWaiting
+                // This test will fail occasionally if we aren't executing atomically.
+                // You can prove this to yourself by replacing `systemUnderTest.async` above with `Task`.
+                XCTAssertFalse(isWaiting)
+                // Signal the semaphore before or after we wait – let the scheduler decide.
+                Task {
+                    semaphore.signal()
+                }
+                // Wait for the concurrent task to complete.
+                await semaphore.wait()
```

Comment:
Here we still need an `await` since `wait()` may suspend. Is that right?

Reply:
Correct! `wait` is an `async` (i.e. possibly suspending) method on the semaphore, whereas `signal` is not. So we need to `await` the `async` method.